contracts/vault/src/execute.rs

```diff
@@ -138,10 +164,20 @@ fn execute_redistribute_funds(deps: DepsMut, env: Env, info: MessageInfo) -> Res
         .load(deps.storage)
         .unwrap_or(Uint128::zero());
 
-    let vault_balance = deps
-        .querier
-        .query_balance(&env.contract.address, &config.usdc_denom)?
-        .amount;
+    let vault_balance = if config.usdc_denom.starts_with("osmo1") {
```

Comment:
Please have a look at how we handled the config in the market contract, which uses an explicit `enum`. This approach could work too, but I'd prefer to have consistency in the implementation if possible.